language/move-prover/doc/user/spec-lang.md

```diff
@@ -819,6 +821,91 @@ fun decrement_ad(addr: address) acquires Counter {
 }
 ```
 
+#### Disabling Invariants
+
+There are times when a global invariant holds almost everywhere,
+except for a brief interval inside a function. In current move code,
+this often occurs when something (e.g. an account) is being setup and
+several structs are published together. Almost everywhere, an
+invariant holds that all of the structs are published or none of them
+are. But the code that publishes the structs must do so sequentially.
+While they are being published, there will be a point where some are
+published and others are not.
+
+In order to verify invariants that hold except during small regions, there
+is a feature to allow users to disable invariants temporarily.
+Consider the following code fragment:
+
+```move
+fn setup() {
+    publish1();
+    publish2();
+    }
+}
+```
+where `publish1` and `publish2` publish two different structs, `T1` and `T2` at address `a`.
+
+```move
+module M {
+    invariant [global] exists<T1>(a) == exists<T2>(a)
+}
+```
+
+As written, the prover will report that the invariant is violated
+after the call to `publish1` and before the call to `publish2`.
+
+To prove that the invariant holds everywhere else, there is a pragma
+```move
+spec setup {
+    pragma disable_invariants_in_body;
+}
+```
+which says that invariants are not required to hold while `setup` is executing.
+
+This pragma changes the Prover's behavior. The invariants are assumed on
+entry to `setup`, but not proved during or after `publish1` and
+`publish2`.  Instead, all invariants that could be invalidated in the
+body of `setup` are asserted and proved at the point of return from
+`setup`.  A consequence of this processing is that the user may need
+to provide stronger post-conditions on `publish1` and `publish2` to
+make it possible to prove the invariants on exit from `setup`.
+
+Another consequence of this processing is that invariants cannot
+safely be assumed to hold during the execution of `publish1` and
+`publish2` (unless nothing in the body of `setup` changes state
+mentioned in the invariant). Therefore, if proving a post-condition
+requires the invariant to be assumed, the post-condition will fail.
+
+In the example, invariants hold at the call sites of `setup`, but not
+in the body. For `publish1`, invariants don't necessarily hold at the
+call site *or* in the body of the function.  In the example, that
+behavior is implied because `publish1` is called in a context where
+invariants are disabled.
```

Comment:
One question a user may have what happens if publish1 is called in a context where invariants are not disabled.

Reply:
It immediately violates the invariant.  I said that explicitly in the revision.